src/query/interpreter.cpp

```diff
@@ -1182,54 +1231,88 @@ PreparedQuery PrepareIndexQuery(ParsedQuery parsed_query, bool in_explicit_trans
-          interpreter_context->db->CreateIndex(label, properties[0]);
+          if (!interpreter_context->db->CreateIndex(label, properties[0])) {
+            index_notification.code = NotificationCode::EXISTANT_INDEX;
+            index_notification.title =
+                fmt::format("Index on label {} on properties {} already exists.", label_name, properties_stringified);
+          }
           EventCounter::IncrementCounter(EventCounter::LabelPropertyIndexCreated);
         }
         invalidate_plan_cache();
       };
       break;
     }
     case IndexQuery::Action::DROP: {
-      handler = [interpreter_context, label, properties = std::move(properties),
-                 invalidate_plan_cache = std::move(invalidate_plan_cache)] {
+      index_notification.code = NotificationCode::DROP_INDEX;
+      index_notification.title = fmt::format("Dropped index on label {} on properties {}.", index_query->label_.name,
+                                             utils::Join(properties_string, ", "));
+      handler = [interpreter_context, label, properties_stringified = std::move(properties_stringified),
+                 label_name = index_query->label_.name, properties = std::move(properties),
+                 invalidate_plan_cache = std::move(invalidate_plan_cache)](Notification &index_notification) {
         if (properties.empty()) {
-          interpreter_context->db->DropIndex(label);
+          if (!interpreter_context->db->DropIndex(label)) {
+            index_notification.code = NotificationCode::NONEXISTANT_INDEX;
+            index_notification.title =
+                fmt::format("Index on label {} on properties {} doesn't exist.", label_name, properties_stringified);
+          }
         } else {
           MG_ASSERT(properties.size() == 1U);
-          interpreter_context->db->DropIndex(label, properties[0]);
+          if (!interpreter_context->db->DropIndex(label, properties[0])) {
+            index_notification.code = NotificationCode::NONEXISTANT_INDEX;
+            index_notification.title =
+                fmt::format("Index on label {} on properties {} doesn't exist.", label_name, properties_stringified);
+          }
         }
         invalidate_plan_cache();
       };
       break;
     }
   }
 
-  return PreparedQuery{{},
-                       std::move(parsed_query.required_privileges),
-                       [handler = std::move(handler)](AnyStream *stream, std::optional<int>) {
-                         handler();
-                         return QueryHandlerResult::NOTHING;
-                       },
-                       RWType::W};
+  return PreparedQuery{
+      {},
+      std::move(parsed_query.required_privileges),
+      [handler = std::move(handler), notifications, index_notification = std::move(index_notification)](
+          AnyStream * /*stream*/, std::optional<int> /*unused*/) mutable {
+        handler(index_notification);
+        notifications->emplace_back(index_notification);
```

Comment:
```suggestion
        notifications->push_back(std::move(index_notification));
```

When only the copy/move constructor needs to be called, I would prefer using `push_back` instead of `emplace_back`, because it eliminates the issue of accidentally calling another constructor (when the type of the vector changes, or when the type of `index_notification` changes).